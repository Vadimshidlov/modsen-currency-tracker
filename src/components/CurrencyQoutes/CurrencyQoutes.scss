@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.quotes__container {
    margin: 0 auto;
    max-width: 1224px;
    margin-bottom: 112px;

    .quotes__title {
        margin-bottom: 24px;
        @include fonts.PoppinsLight;
        color: colors.$color_white;
        font-size: fonts.$font-size_32;
        line-height: 41.14px;
    }

    .title__border {
        margin-bottom: 50px;

        background-color: colors.$currency_card_border;
        height: 2px;
        width: 520px;
    }

    .quotes__items {
        display: flex;
        column-gap: 184px;
        row-gap: 56px;
        flex-wrap: wrap;
    }
}
