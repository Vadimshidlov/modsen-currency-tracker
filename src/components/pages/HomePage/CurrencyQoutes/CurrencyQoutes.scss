@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;

.quotes__container {
    max-width: 1224px;
    margin-bottom: 96px;

    @media (max-width: 500px) {
        margin-bottom: 151px;
    }

    .quotes__title {
        margin-bottom: 24px;
        @include fonts.PoppinsLight;
        color: colors.$color_white;
        font-size: fonts.$font-size_32;
        line-height: 41.14px;

        @media (max-width: 768px) {
            margin-bottom: 12px;
            font-size: fonts.$font-size_24;
            line-height: 22px;
        }

        @media (max-width: 500px) {
            margin-bottom: 12px;
            font-size: fonts.$font-size_18;
            line-height: 17px;
        }

        @media (max-width: 450px) {
            margin-bottom: 7px;
            font-size: fonts.$font-size_12;
            line-height: 15px;
        }
    }

    .title__border {
        margin-bottom: 50px;
        background-color: colors.$currency_card_border;
        height: 2px;
        width: 520px;

        @media (max-width: 1120px) {
            margin-bottom: 35px;
            width: 100%;
        }

        @media (max-width: 500px) {
            margin-bottom: 16px;
            width: 100%;
        }
    }

    .quotes__items {
        display: flex;
        justify-content: space-between;
        row-gap: 56px;
        flex-wrap: wrap;

        @media (max-width: 1120px) {
            display: flex;
            flex-direction: column;
            row-gap: 25px;
        }

        @media (max-width: 768px) {
            display: flex;
            flex-direction: column;
            row-gap: 16px;
        }
    }

    &.light {
        .quotes__title {
            color: colors.$black_background;
        }

        .currency-card__container {
            background: colors.$currency-card_background__light;
        }

        .currency-card__container {
            .currency-card__titile {
                color: colors.$black_background;
            }

            .currency-card__percent {
                color: colors.$black_background;
            }
        }
    }
}
