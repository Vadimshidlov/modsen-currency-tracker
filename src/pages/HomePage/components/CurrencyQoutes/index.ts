export { default } from "@/pages/HomePage/components/CurrencyQoutes/CurrencyQoutes";
