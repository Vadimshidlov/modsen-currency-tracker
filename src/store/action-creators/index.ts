export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
