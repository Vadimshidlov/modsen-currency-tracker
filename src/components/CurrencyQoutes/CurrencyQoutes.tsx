import React from "react";
import Text from "@/components/Text/Text";
import CurrencyCard from "@/components/CurrencyCard/CurrencyCard";
import "@/components/CurrencyQoutes/CurrencyQoutes.scss";

export default function CurrencyQoutes() {
    return (
        <div className="quotes__container">
            <Text className="quotes__title title">Quotes</Text>
            <div className="title__border" />
            <div className="quotes__items">
                {[1, 2, 3, 4, 5, 6, 7, 8, 9].map(() => (
                    <CurrencyCard />
                ))}
            </div>
        </div>
    );
}
