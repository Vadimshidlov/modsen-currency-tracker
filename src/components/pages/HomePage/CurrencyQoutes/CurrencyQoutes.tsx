import React from "react";
import Text from "@/components/pages/HomePage/Text/Text";
import CurrencyCard from "@/components/pages/HomePage/CurrencyCard/CurrencyCard";
import "@/components/pages/HomePage/CurrencyQoutes/CurrencyQoutes.scss";
import { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";
import ModalWindow from "@/components/pages/HomePage/ModalWindow/ModalWindow";

export default function CurrencyQoutes() {
    const { currency } = useTypedSelectorHook((state) => state.currency);
    const { isOpen } = useTypedSelectorHook((state) => state.modalWindow);

    return (
        <div className="quotes__container">
            <Text className="quotes__title title">Quotes</Text>
            <div className="title__border" />
            <div className="quotes__items">
                {Object.keys(currency.data).map((element) => (
                    <CurrencyCard
                        currencyCode={element}
                        key={element}
                        value={currency.data[element].value}
                    />
                ))}
            </div>
            <ModalWindow isOpen={isOpen} />
        </div>
    );
}
