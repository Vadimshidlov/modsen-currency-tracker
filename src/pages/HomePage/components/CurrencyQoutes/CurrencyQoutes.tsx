import React from "react";
import Text from "@/pages/HomePage/components/Text/index";
import CurrencyCard from "@/pages/HomePage/components/CurrencyCard/index";
import "@/pages/HomePage/components/CurrencyQoutes/CurrencyQoutes.scss";
import { useTypedSelectorHook } from "@/store/hooks/index";
import ModalWindow from "@/pages/HomePage/components/ModalWindow/index";
import { selectCurrency, selectModal, selectTheme } from "@/store/selectors/index";

export default function CurrencyQoutes() {
    const { currency } = useTypedSelectorHook(selectCurrency);
    const { isOpen } = useTypedSelectorHook(selectModal);
    const { theme } = useTypedSelectorHook(selectTheme);

    return (
        <section className={theme === "Light" ? "quotes__container light" : "quotes__container"}>
            <Text className="quotes__title title">Quotes</Text>
            <div className="title__border" />
            <div className="quotes__items">
                {Object.keys(currency.data).map((element) => (
                    <CurrencyCard
                        currencyCode={element}
                        key={element}
                        value={currency.data[element].value}
                    />
                ))}
            </div>
            <ModalWindow isOpen={isOpen} />
        </section>
    );
}
