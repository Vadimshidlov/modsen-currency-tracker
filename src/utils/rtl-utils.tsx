import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
