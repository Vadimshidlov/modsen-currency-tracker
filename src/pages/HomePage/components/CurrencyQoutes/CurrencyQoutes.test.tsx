import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import CurrencyQoutes from "@/pages/HomePage/components/CurrencyQoutes/index";
import { store } from "@/store";

const mockResponse = {
    meta: {
        last_updated_at: "2024-03-04T23:59:59Z",
    },
    data: {
        USD: { code: "USD", value: 1 },
        EUR: { code: "EUR", value: 1 },
        BTC: { code: "BTC", value: 1 },
        JPY: { code: "JPY", value: 1 },
    },
};

describe("CurrencyQoutes component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<CurrencyQoutes />);

        expect(screen.getByText("Quotes")).toBeInTheDocument();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(<CurrencyQoutes />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <CurrencyQoutes />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("should correct render component with correct preloaded state", async () => {
        renderWithProviders(<CurrencyQoutes />, {
            preloadedState: {
                currency: {
                    currency: mockResponse,
                    isLoading: false,
                    error: null,
                },
            },
        });

        expect(screen.getByText("Commercial Dollar")).toBeInTheDocument();
        expect(screen.getByText("Euro")).toBeInTheDocument();
        expect(screen.getByText("Bitcoin")).toBeInTheDocument();
        expect(screen.getByText("Yen")).toBeInTheDocument();
    });
});
