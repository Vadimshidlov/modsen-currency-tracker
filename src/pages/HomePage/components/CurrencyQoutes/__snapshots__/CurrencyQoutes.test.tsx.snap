// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyQoutes component tests should match snapshot 1`] = `
<section
  className="quotes__container"
>
  <p
    className="quotes__title title"
  >
    Quotes
  </p>
  <div
    className="title__border"
  />
  <div
    className="quotes__items"
  />
</section>
`;

exports[`CurrencyQoutes component tests snapshot matches 1`] = `
<DocumentFragment>
  <section
    class="quotes__container"
  >
    <p
      class="quotes__title title"
    >
      Quotes
    </p>
    <div
      class="title__border"
    />
    <div
      class="quotes__items"
    />
  </section>
</DocumentFragment>
`;
